index.js

```diff
@@ -1,13 +1,117 @@
-//
+/* */
+const BUILD_MODE = typeof window === 'undefined'
+import Core from 'css-modules-loader-core'
+import path from 'path'
+import autoprefixer from './autoprefixer'
+let _source = []
 
-import {Plugins} from './lib/plugins'
-import {CSSLoader} from './lib/CSSLoader'
+class CSSLoader {
+  constructor( plugins, moduleName ) {
+    this.fetch = this.fetch.bind( this )
+    this.bundle = this.bundle.bind( this )
+    this.moduleName = moduleName || __moduleName
+    this.core = new Core( plugins )
+    this._cache = { _source }
+    this._deps = {}
+  }
 
-const {fetch, bundle} = new CSSLoader([
+  fetch( load, fetch ) {
+    let path = load.metadata.pluginArgument,
+      deps = this._deps[path] = []
+    // Use the default Load to fetch the source
+    return fetch( load ).then( source => {
+      // Pass this to the CSS Modules core to be translated
+      // triggerImport is how dependencies are resolved
+      return this.core.load( source, path, "A", this.triggerImport.bind( this ) )
+    } ).then( ( { injectableSource, exportTokens } ) => {
+      if ( BUILD_MODE ) {
+        this._cache["./" + path] = exportTokens
+        _source.push( injectableSource )
+        return `export default ${JSON.stringify( exportTokens )}`
+      } else {
+        // Once our dependencies are resolved, inject ourselves
+        this.createElement( injectableSource, path )
+        // And return out exported variables.
+        let imports = deps.map( d => `import "${d}"` ).join( ";" ),
+        // on a reload, the only thing we need to do is cause a repaint
+          hotReloader = `export let __hotReload = () => document.body.offsetHeight && true`,
+          exports = `export default ${JSON.stringify( exportTokens )}`
+        return [imports, hotReloader, exports].join( ";" )
```

Comment:
The `imports` and `hotReloader` here are the key, I think. Without the imports, SystemJS doesn't know that file `a.css` depends on `b.css`. Without the `hotReloader`, `jspm-server` doesn't know it's safe to reload this module.